src/software/ai/hl/stp/tactic/shadow_enemy/shadow_enemy_fsm.h

```diff
@@ -0,0 +1,215 @@
+     * @param enemyTeam The enemy team
+     * @param shadowee The enemy robot we are shadowing
+     * @param shadow_distance The distance our friendly robot will position itself away
+     * from the shadowee
+     */
+    static Point findBlockShotPoint(const Robot &robot, const Field &field,
+                                    const Team &friendlyTeam, const Team &enemyTeam,
+                                    const Robot &shadowee, const double &shadow_distance)
+    {
+        std::vector<Robot> robots_to_ignore = {robot};
+        if (friendlyTeam.goalie().has_value())
+        {
+            robots_to_ignore.emplace_back(friendlyTeam.goalie().value());
+        }
+
+        auto best_enemy_shot_opt =
+            calcBestShotOnGoal(field, friendlyTeam, enemyTeam, shadowee.position(),
+                               TeamType::FRIENDLY, robots_to_ignore);
+
+        Vector enemy_shot_vector = field.friendlyGoalCenter() - shadowee.position();
+        if (best_enemy_shot_opt)
+        {
+            enemy_shot_vector =
+                best_enemy_shot_opt->getPointToShootAt() - shadowee.position();
+        }
+        return shadowee.position() + enemy_shot_vector.normalize(shadow_distance);
+    }
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto block_shot_s     = state<MoveFSM>;
+        const auto block_pass_s     = state<BlockPassState>;
+        const auto steal_and_chip_s = state<StealAndChipState>;
+
+        const auto update_e = event<Update>;
+
+        /**
+         * Guard that checks if the enemy threat has ball
+         *
+         * @param event ShadowEnemyFSM::Update
+         *
+         * @return if the ball has been have_possession
+         */
+        const auto enemy_threat_has_ball = [](auto event) {
+            std::optional<EnemyThreat> enemy_threat_opt =
+                event.control_params.enemy_threat;
+            if (enemy_threat_opt.has_value())
+            {
+                return enemy_threat_opt.value().has_ball;
+            };
+            LOG(WARNING) << "Enemy threat not initialized for robot "
+                         << event.common.robot.id() << "\n";
+            return false;
+        };
+
+        /**
+         * Action to block the pass to our shadowee
+         *
+         *
+         * @param event ShadowEnemyFSM::Update
+         */
+        const auto block_pass = [](auto event) {
+            std::optional<EnemyThreat> enemy_threat_opt =
+                event.control_params.enemy_threat;
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point position_to_block =
+                ball_position +
+                (event.common.world.field().friendlyGoalCenter() - ball_position)
+                    .normalize(event.control_params.shadow_distance);
```

Comment:
Can we add comments so we're explicit that this is a default blocking position if we don't have an enemy_threat?